Documentation/project-docs/writing-tests.md

```diff
@@ -0,0 +1,93 @@
+This document provides an overview of paradigm we are using in our tests.
+
+# RemoteExecutor
```

Comment:
/cc @AlexGhiondea 

I've found that this is in general a bad pattern:
1. Exceptions are not properly marshalled back and when the app crashes we get no relevant information.
2. This currently blocks UWP testing (internal thread since we're still working on the tooling) and there is no foreseeable fix for the problem.
3. The exe is written as a .Net Framework application (not sure how well it actually tests CoreFX)